src/Core/Grid/Definition/Factory/CustomerAddressGridDefinitionFactory.php

```diff
@@ -54,7 +54,19 @@ final class CustomerAddressGridDefinitionFactory extends AbstractGridDefinitionF
     public function __construct(HookDispatcherInterface $hookDispatcher, ?Request $currentRequest)
     {
         parent::__construct($hookDispatcher);
-        $this->backUrl = $currentRequest ? $currentRequest->getUri() : '';
+
+        // Remove useless query parameters to avoid some strange behaviors with delete workflow
+        // This way, behavior of "customer address grid" is consistent with `AddressGridDefinitionFactory`
+        // (should be solved using proper front-end and API architecture)
+
+        $query = $currentRequest->query;
+        $query->remove('customer_address');
+        $qs = http_build_query($query->all(), '', '&');
+
+        $server = $currentRequest->server;
+        $server->set('QUERY_STRING', $qs);
```

Comment:
I see no reason to alter the `QUERY_STRING` in `$_SERVER` as only the `backUrl` is needed.
Why you don't try to change the `query` then reuse the `$currentRequest->getUri()`?

Reply:
It was my first intent but exploring Symfony implementation, i discovered that `server` attributes is used in most of URI / sub URI constructions from Request object ...

Take a look to [getUri()](https://github.com/symfony/symfony/blob/v4.4.39/src/Symfony/Component/HttpFoundation/Request.php#L1027) and [getQueryString()](https://github.com/symfony/symfony/blob/v4.4.39/src/Symfony/Component/HttpFoundation/Request.php#L1111)

I think ending by stupid parsing/concatenating shows that Symfony class is a really bad implementation of object oriented class that should represent an Http Request request ...